src/Discord/Parts/Channel/Poll/PollAnswer.php

```diff
@@ -0,0 +1,172 @@
+
+use Discord\Helpers\Collection;
+use Discord\Http\Endpoint;
+use Discord\Parts\Channel\Channel;
+use Discord\Parts\Channel\Message;
+use Discord\Parts\Guild\Guild;
+use Discord\Parts\Part;
+use Discord\Parts\Thread\Thread;
+use Discord\Parts\User\User;
+use React\Promise\ExtendedPromiseInterface;
+
+/**
+ * An answer to a poll.
+ *
+ * @link https://discord.com/developers/docs/resources/poll#poll-answer-object
+ *
+ * @since 10.0.0
+ *
+ * @property int        $answer_id   The ID of the answer. Only sent as part of responses from Discord's API/Gateway.
+ * @property PollMedia  $poll_media  The data of the answer
+ *
+ * @property      string         $user_id    The user ID that voted for the answer.
+ * @property-read User           $user       The user that voted for the answer.
+ * @property      string         $channel_id The channel ID that the poll belongs to.
+ * @property-read Channel|Thread $channel    The channel that the poll belongs to.
+ * @property      string         $message_id The message ID that the poll belongs to.
+ * @property      Message|null   $message    The message the poll belongs to.
+ * @property      string|null    $guild_id   The guild ID of the guild that owns the channel the poll message is in.
+ * @property-read Guild|null     $guild      The guild that owns the channel the poll belongs in.
+ */
+class PollAnswer extends Part
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected $fillable = [
+        'answer_id',
+        'poll_media',
+
+        // events
+        'user_id',
+        'channel_id',
+        'message_id',
+        'guild_id',
+    ];
+
+    /**
+     * Gets the user attribute.
+     *
+     * @return User
+     */
+    protected function getUserAttribute(): User
+    {
+        return $this->discord->users->get('id', $this->user_id);
+    }
+
+    /**
+     * Gets the channel attribute.
+     *
+     * @return Channel|Thread
+     */
+    protected function getChannelAttribute(): Part
+    {
+        if ($guild = $this->guild) {
+            $channels = $guild->channels;
+
+            if ($channel = $channels->get('id', $this->channel_id)) {
+                return $channel;
+            }
+
+            foreach ($channels as $parent) {
+                if ($thread = $parent->threads->get('id', $this->channel_id)) {
+                    return $thread;
+                }
+            }
+        }
+
+        // @todo potentially slow
+        if ($channel = $this->discord->getChannel($this->channel_id)) {
+            return $channel;
+        }
+
+        return $this->factory->part(Channel::class, [
+            'id' => $this->channel_id,
+            'type' => Channel::TYPE_DM,
+        ], true);
+    }
+
+    /**
+     * Gets the message attribute.
+     *
+     * @return Message|null
+     */
+    protected function getMessageAttribute(): ?Message
+    {
+        if ($channel = $this->channel) {
+            return $channel->messages->get('id', $this->message_id);
```

Comment:
This is tending to always return `null` inside of the `MESSAGE_POLL_VOTE_ADD` and `MESSAGE_POLL_VOTE_REMOVE` events. It would be nice if there was a way to reliably fix it so the `poll` instance could be fetched easily.

Reply:
This is indeed similar structure to Message Reactions, so i expect the code should be similar to Reaction codes as well. Also please mark which fillables are not part of the actual object, i.e. events with `//events` and always put it last (before a nested repository)

Reply:
I suspect once the message cache is fixed, this will be fixed – going to resolve this in favor of https://github.com/discord-php/DiscordPHP/pull/1226#discussion_r1685747340 and https://github.com/discord-php/DiscordPHP/pull/1226#discussion_r1685747340